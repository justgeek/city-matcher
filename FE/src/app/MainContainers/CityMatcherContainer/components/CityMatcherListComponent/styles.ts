import styled from "styled-components";

export const StyledCityMatcherComponent = styled.div`
  display: flex;
  flex-direction: column;
  height: 100%;
  overflow-y: scroll;
`;
