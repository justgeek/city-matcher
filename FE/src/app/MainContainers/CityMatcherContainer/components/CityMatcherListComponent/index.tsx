import * as React from "react";
import { City } from "../../reducer";
import CityRowComponent from "./components/CityRowComponent";
import { fetchOldCitiesPayload, logCityPayload } from "../../actions";
import { StyledCityMatcherComponent } from "./styles";

interface Props {
  cities: City[];
  autoCompleteResults: City[];
  onCityInputChange: ({ query }: fetchOldCitiesPayload) => void;
  onCityInputFocus: (index) => void;
  activateRow: (index) => void;
  logCity: ({ cityLog }: logCityPayload) => void;
  currentActiveRow: number | null;
  currentQueryPage: number | null;
}
export const NUMBER_OF_RESULTS = 10; // how many results are shown per page;

// Component DOM
export default class CityMatcherListComponent extends React.PureComponent<
  Props
> {
  private renderCityRow = (city: City, index: number) => {
    const {
      onCityInputChange,
      onCityInputFocus,
      activateRow,
      currentActiveRow,
      autoCompleteResults,
      logCity
    } = this.props;
    const isActive = currentActiveRow === index;
    return (
      <CityRowComponent
        key={index}
        city={city}
        autoCompleteResults={autoCompleteResults}
        onCityInputChange={onCityInputChange}
        onCityInputFocus={onCityInputFocus}
        activateRow={activateRow}
        index={index}
        isActive={isActive}
        logCity={logCity}
      />
    );
  }

  private sliceCitiesPerPage(cities, page) {
    const startOfSlice = (page - 1) * NUMBER_OF_RESULTS;
    const endOfSlice = startOfSlice + NUMBER_OF_RESULTS + 1;

    return cities.slice(startOfSlice, endOfSlice);
  }
  render() {
    const { cities, currentQueryPage } = this.props;
    return (
      <StyledCityMatcherComponent>
        <table className="uk-table uk-table-divider uk-table-striped">
          <tbody>
            {this.sliceCitiesPerPage(cities, currentQueryPage).map(
              this.renderCityRow
            )}
          </tbody>
        </table>
      </StyledCityMatcherComponent>
    );
  }
}
