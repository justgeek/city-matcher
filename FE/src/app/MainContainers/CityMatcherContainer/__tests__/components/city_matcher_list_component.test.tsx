import CityMatcherListComponent from "MainContainers/CityMatcherContainer/components/CityMatcherListComponent";
import { shallow } from "enzyme";
import { cities } from "../../__mocks__/cities.mock";
import { autoCompleteResults } from "../../__mocks__/autocomplete_results.mock";
import toJson from "enzyme-to-json";
import * as React from "react";

describe("<CityMatcherListComponent />", () => {
  describe("render()", () => {
    test("renders CityMatcherListComponent", () => {
      const onCityInputChange = jest.fn();
      const onCityInputFocus = jest.fn();
      const activateRow = jest.fn();
      const logCity = jest.fn();
      const wrapper = shallow(
        <CityMatcherListComponent
          cities={cities}
          autoCompleteResults={autoCompleteResults}
          onCityInputChange={onCityInputChange}
          onCityInputFocus={onCityInputFocus}
          activateRow={activateRow}
          logCity={logCity}
          currentActiveRow={1}
          currentQueryPage={8}
        />
      );
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});
