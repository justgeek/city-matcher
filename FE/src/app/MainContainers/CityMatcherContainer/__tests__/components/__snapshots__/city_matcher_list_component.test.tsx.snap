// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CityMatcherListComponent /> render() renders CityMatcherListComponent 1`] = `
<div
  className="sc-htpNat fczHYf"
>
  <table
    className="uk-table uk-table-divider uk-table-striped"
  >
    <tbody />
  </table>
</div>
`;
